src/Core/SortCursor.h

```diff
@@ -190,8 +225,26 @@ struct SimpleSortCursor : SortCursorHelper<SimpleSortCursor>
         const auto & desc = impl->desc[0];
         int direction = desc.direction;
         int nulls_direction = desc.nulls_direction;
-        int res = impl->sort_columns[0]->compareAt(lhs_pos, rhs_pos, *(rhs.impl->sort_columns[0]), nulls_direction);
-        return res != 0 && ((res > 0) == (direction > 0));
+
+        bool result = false;
+
+#if USE_EMBEDDED_COMPILER
+        if (impl->desc.compiled_sort_description && rhs.impl->desc.compiled_sort_description)
```

Comment:
Should `impl->desc.compiled_sort_description` and `rhs.impl->desc.compiled_sort_description` be the same by construction? The `rhs.impl->desc.compiled_sort_description` is not used below, but there's no difference which instance to use, is it?